argocd/provider.go

```diff
@@ -278,6 +283,7 @@ func argoCDProviderConfigFromResourceData(ctx context.Context, d *schema.Resourc
 		PlainText:                getBoolFromResourceData(d, "plain_text"),
 		PortForward:              getBoolFromResourceData(d, "port_forward"),
 		PortForwardWithNamespace: getStringFromResourceData(d, "port_forward_with_namespace"),
+		ServerName:               getStringFromResourceData(d, "server_name"),
```

Comment:
``` suggestion
```

I'm afraid this is not going to have the desired behaviour. What you want to be doing is setting it [here](https://github.com/argoproj-labs/terraform-provider-argocd/blob/dfa0f72848fc1bb90f6ffe8cc9f943dd0f7c92e6/internal/provider/model_provider.go#L239) instead. 

Something like the following should do the trick:
```go
opts.ServerName = getDefaultString(p.ServerName, "ARGOCD_SERVER_NAME")

if opts.ServerName == "" {
	opts.ServerName = "argocd-server"
}
```